keepassxc-browser/content/observer-helper.js

```diff
@@ -270,3 +282,17 @@ kpxcObserverHelper.ignoredNode = function(target) {
 
     return false;
 };
+
+// Traverses all children, including Shadow DOM elements
+const traverseChildren = function(target, children) {
+    for (const child of target.childNodes) {
+        if (child.nodeName === 'INPUT' && child.type !== 'hidden' && !child.disabled ) {
+            children.push(child);
+        }
+
+        traverseChildren(child, children);
+        if (child.shadowRoot) {
+            traverseChildren(child.shadowRoot, children);
+        }
```

Comment:
We should set a traversal limit, perhaps 100 levels?

Reply:
Fixed.

Reply:
No that doesn't solve it, that makes this function do weird things. I meant a recursion limit (depth). I feel like we are reimplementing a basic jQuery function here. Finding specific node types isn't a built in feature of the DOM?

Reply:
Using that function is not allowed for a `NodeList`, so I'll need to make some more modifications. Also the function calls itself over and over, so a different kind of counter needs to be implemented.

Reply:
You'd have to add a parameter to the function that tracks depth, defaults to 0. You then check if the max depth has been reached and stop the recursion if so.

Reply:
Count added. Needed to revert some changes because we actually need to traverse non-input elements also. But scripts etc. must be ignored, and this is why the extra `ignoredElement()` check is added.